lib/parser/fileParser.ts

```diff
@@ -29,6 +29,9 @@ export class FileParser extends ParserBase {
     // In the end lexerTokens === this.file.lexerTokens === this.lexerTokens
     const lexerTokens: [] = [];
     this.file = new OFile(this.text, this.state.fileUri, this.originalText, lexerTokens);
+    if (this.text.length > settings.analysis.maxFileSize * 1024) {
```

Comment:
In the project parser we do the check before reading the file, do you think think this makes a noticable difference?

Reply:
Also to be nitpicky, the string size is not actually the file size, because of utf8 :clown_face:

Reply:
Yes, in the project parser the actual file length is used and not the string length.
I believe, reading the file in this case is fine because the fileParser will only be called if a user opens the file directly (the projectParser avoids the parser altogether)